src/components/DelegateMultipleButton.jsx

```diff
@@ -0,0 +1,363 @@
+  }
+
+  componentDidMount() {
+    this.dacsObserver = feathersClient
+      .service('dacs')
+      .watch({ listStrategy: 'always' })
+      .find({
+        query: {
+          delegateId: { $gt: '0' },
+          ownerAddress: this.props.currentUser.address,
+          $select: ['ownerAddress', 'title', '_id', 'delegateId', 'delegateEntity', 'delegate'],
+        },
+      })
+      .subscribe(
+        resp => {
+          const dacs = resp.data.map(c => ({
+            name: c.title,
+            id: c._id,
+            ownerAddress: c.ownerAddress,
+            delegateId: c.delegateId,
+            delegateEntity: c.delegateEntity,
+            delegate: c.delegate,
+            type: 'dac',
+          }));
+
+          const delegationOptions = this.props.milestone
+            ? dacs.concat([
+                {
+                  id: this.props.milestone.campaign._id,
+                  name: this.props.milestone.campaign.title,
+                  projectId: this.props.milestone.campaign.projectId,
+                  ownerEntity: this.props.milestone.ownerEntity,
+                  type: 'campaign',
+                },
+              ])
+            : dacs;
+
+          this.setState({ delegationOptions }, () => {
+            if (delegationOptions.length === 1) {
+              this.selectedObject({ target: { value: [delegationOptions[0].id] } });
+            }
+          });
+        },
+        () => {},
+      );
+  }
+
+  selectedObject({ target }) {
+    this.setState({ objectToDelegateFrom: target.value, isLoadingDonations: true });
+    this.loadDonations(target.value);
+  }
+
+  loadDonations(ids) {
+    if (ids.length !== 1) return;
+
+    const entity = this.state.delegationOptions.find(c => c.id === ids[0]);
+
+    if (this.donationsObserver) this.donationsObserver.unsubscribe();
+
+    const options = {};
+
+    switch (entity.type) {
+      case 'dac':
+        options.delegateId = entity.delegateId;
+        options.delegateTypeId = entity.id;
+        options.status = Donation.WAITING;
+
+        break;
+      case 'campaign':
+        options.ownerId = entity.projectId;
+        options.ownerTypeId = entity.id;
+        options.status = Donation.COMMITTED;
+        break;
+      default:
+        break;
+    }
+
+    const query = paramsForServer({
+      query: {
+        amountRemaining: { $ne: 0 },
+        ...options,
+        $sort: { createdAt: 1 },
+      },
+      schema: 'includeTypeAndGiverDetails',
+    });
+
+    // start watching donations, this will re-run when donations change or are added
+    this.donationsObserver = feathersClient
+      .service('donations')
+      .watch({ listStrategy: 'always' })
+      .find(query)
+      .subscribe(
+        r => {
+          const delegations = r.data.map(d => new Donation(d));
+          let amount = utils.fromWei(
+            delegations
+              .reduce((sum, d) => sum.plus(new BigNumber(d.amountRemaining)), new BigNumber('0'))
```

Comment:
just an fyi, but if you use utils.toBN(..) instead of new BigNumber, you can pass the BN object to fromWei & don't have to call .toString()